src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -212,7 +212,7 @@ public IndexShard(ShardId shardId, @IndexSettings Settings indexSettings, IndexS
         /* create engine config */
 
         this.config = new EngineConfig(shardId,
-                indexSettings.getAsBoolean(EngineConfig.INDEX_OPTIMIZE_AUTOGENERATED_ID_SETTING, true),
```

Comment:
I'd want to remove this entirely but this change is supposed to go into 1.4 as well afaik so I guess that is fine.